app/src/main/java/com/xmartlabs/gong/ui/screens/signin/SignInScreenActionEventState.kt

```diff
@@ -0,0 +1,24 @@
+package com.xmartlabs.gong.ui.screens.signin
+
+import androidx.compose.runtime.Immutable
+
+/**
+ * Created by mirland on 22/4/21.
+ */
+sealed class SignInViewModelEvent {
+  object NavigateToDashboard : SignInViewModelEvent()
+  class SignInError(val throwable: Throwable) : SignInViewModelEvent()
+}
+
+sealed class SignInUiAction {
+  object SignIn : SignInUiAction()
+  data class ChangePassword(val password: String) : SignInUiAction()
+  data class ChangeUserName(val userName: String) : SignInUiAction()
+}
```

Comment:
Maybe we can have the children classes outside so we don't have to reference them as part of SignInUiAction as it is quite long and that way they are available in the package

Reply:
I'm not sure, in a real app each screen will define actions, so you have to find your specific action easily. Suppose that you have multiple screens that can change the username, for example, signUp, sign in, forgot password, and so on. In that case, you should be able to use your specific class easily. So if we don't define it inside the `SignInUiAction` we would rename it as `ChangeUserNameSignInAction` and I think it can be worst. Furthermore, it will be harder to find it, right now if you write `SignInUiAction.` the android studio will display all available actions, but if you don't use it in this way, you won't be able to use this feature. what do you think?

Reply:
Maybe make them package private, so it only can be used by the intended class and you don't have to rename them.
Of course, this is thinking that the actions from the screens are never shared

Reply:
but you cannot do it in kotlin right? you can make the actions internal in your module, but it's not the same, they will be accessible in all classes and you will have all problems that I mentioned